Science/src/temp_humidity/temp_humidity.cpp

```diff
@@ -1,23 +1,57 @@
 #include "temp_humidity.h"
 
-TempHumiditySensor::TempHumiditySensor() : sensor(&Wire, 0x44, 4) { }
+void TempHumiditySensor::readReg(){
+  Wire.beginTransmission(addr);
+  // Rewrote with overloaded method so that you don't need to use & on a literal
+  Wire.write(reg);
+  uint8_t code = Wire.endTransmission();
+  if(code != 0){
+    // Wiping buffer if there is an error
+    memset(buf, 0, 4);
+    Serial.print("Error: Could not read temperature/humidity (Error code ");
+    Serial.print(code);
+    Serial.println(")");
+    return;
+  }
+
+  delay(20);
+  
+  // Requesting 4 bytes to get data from registers 0x00 and 0x01 and putting it in buf[0:4]
+  Wire.requestFrom(addr, 4);
+  for(int i = 0; i < 4; i++){
+    buf[i] = Wire.read();
+  }
+  return;
+}
+
+void TempHumiditySensor::setup(){ 
+  // 5 second delay before issuing error
+  unsigned long startTime = millis();
+  unsigned long timeout = 5000;
 
-void TempHumiditySensor::setup() {
   Serial.print("Initializing temp/humidity sensor...");
-  while (sensor.begin() != 0) {
+  while (Wire.begin() != 1) {
     Serial.print(".");
+     if (millis() - startTime > timeout) {
+        // Timeout reached, so issue an error
+        Serial.println("Sensor setup failed.");
```

Comment:
Careful when leaving `setup()` early -- the rest of the code assumes that `setup()` finished without problems. What happens to `readReg()`'s `Wire.beginTransmission()`, `Wire.write()`, and `Wire.requestFrom()` if `Wire.begin()` did not finish? I see you did check `Wire.endTransmission()`, so if that is enough then good job there. 

What I usually do here is add a field to the class `bool isReady = false`, then set that to true at the end of the `setup()` function. Then, all other functions can do a quick `if (!isReady) return;` to avoid getting into undefined behavior